src/Microsoft.TemplateEngine.Edge/Settings/TemplateInfo.cs

```diff
@@ -31,18 +32,86 @@ public TemplateInfo(JObject entry)
             GroupIdentity = entry[nameof(GroupIdentity)].ToString();
             Name = entry[nameof(Name)].ToString();
             ShortName = entry[nameof(ShortName)].ToString();
-            JObject tagsObject = (JObject) entry[nameof(Tags)];
-            Dictionary<string, string> tags = new Dictionary<string, string>();
-            Tags = tags;
-            //using (Timing.Over("Read tags"))
-                foreach (JProperty item in tagsObject.Properties())
+
+            // parse the cached tags
+            Dictionary<string, ICacheTag> tags = new Dictionary<string, ICacheTag>();
+            JObject tagsObject = (JObject)entry[nameof(Tags)];
+            foreach (JProperty item in tagsObject.Properties())
+            {
+                Dictionary<string, string> choicesAndDescriptions = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+                JObject cdToken = (JObject)item.Value[nameof(ICacheTag.ChoicesAndDescriptions)];
+                foreach (JProperty cdPair in cdToken.Properties())
```

Comment:
This should probably use the helper to make sure we're reading case insensitively & have null checks in case we're reading a cache that doesn't have this new property